openmsupply/application-manager-server
# #475 Updates to OrgJoin template

This pretty much completes [Epic #42 - Join organisation](https://github.com/openmsupply/application-manager-web-app/issues/475)

Requires front-end branch from [PR#498](https://github.com/openmsupply/application-manager-web-app/pull/498)

Now the Company selection in the front-end is carried through and applied correctly to join the user to organisation in the back-end on Review approval.

### To test:

(Requires fresh `database_init`)
1. Start a new OrgJoin application: http://localhost:3000/application/new?type=OrgJoin
2. Fill out and Submit application
3. Assign the user "Screener1" to this application using the following SQL code:  
```
UPDATE review_assignment SET status = 'Assigned' WHERE id = 1;
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '5002', '1');
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '43', '1');
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '42', '1');
INSERT INTO public.review_question_assignment (id, template_element_id, review_assignment_id) VALUES (DEFAULT, '39', '1');
```
4. As "Screener1", start review and approve all questions. Submit review.
5. Back-end should correctly update Status, Outcome, create a new user_organisation record, and grant "ApplyDrugReg" permissions to the user/org


Unrelated changes: added some missing documentation for "CleanupFiles" action
